### DISCUSSION TITLE
Issue with Transfer Samples in Connector Tutorial

### DISCUSSION THREAD

**Discussion by anstrasser** [eclipse-edc/Samples#190] [Q&A] [answered]
Hello everyone,

I'm currently a student working on my thesis about dataspaces, and I'm going through the Connector Tutorial. I'm encountering difficulties in Step 5 of the "Transfer Sample 01 Negotiation." As per the instructions, the following cURL command should be used:

```
curl -d @transfer/transfer-01-negotiation/resources/negotiate-contract.json \
  -X POST -H 'content-type: application/json' http://localhost:29193/management/v2/contractnegotiations \
  -s | jq
```

The response I receive looks like this:

```
<html>
<head>
    <meta http-equiv="Content-Type" content="text/html;charset=ISO-8859-1" />
    <title>Error 400 Bad Request</title>
</head>
<body>
    <h2>HTTP ERROR 400 Bad Request</h2>
    <table>
        <tr>
            <th>URI:</th>
            <td>/management/v2/contractnegotiations</td>
        </tr>
        <tr>
            <th>STATUS:</th>
            <td>400</td>
        </tr>
        <tr>
            <th>MESSAGE:</th>
            <td>Bad Request</td>
        </tr>
        <tr>
            <th>SERVLET:</th>
            <td>EDC-management</td>
        </tr>
    </table>
</body>
</html>
```

I suspect the issue might be with the negotiate-contract.json file, specifically with "@id": "{{contract-negotiation-id}}". It seems that this ID may not be known at this stage. Could someone help me overcome Step 5 of the Transfer Sample and successfully complete the negotiation?

Thanks in advance for your assistance!

Andy

P.S.: I have a suspicion that there might be a missing step before Step 5, but I'm not entirely sure.

**Comment by ndr-brt** [comment, marked as answer]
Yes, you need to replace the placeholder with the actual contract offer it (not negotiation).
That was described in the sample script but it did get eventually lost, will provide a PR to fix that.

**Reply by anstrasser** [reply]
Hi,
I've gone through the tutorial again from the beginning, with the changes you mad. However, I'm still encountering the same error. Additionally, in the Provider Terminal, I see:
`WARNING 2024-01-17T09:43:22.8336145 Error expanding JSON-LD structure
JsonLdError[code=There was a problem encountered loading a remote context [code=LOADING_REMOTE_CONTEXT_FAILED]., message=There was a problem encountered loading a remote context [code=LOADING_REMOTE_CONTEXT_FAILED].]
(...)
Caused by: JsonLdError[code=The document could not be loaded or parsed [code=LOADING_DOCUMENT_FAILED]., message=URL scheme [http] is not supported.]`

Could you please assist me further in resolving this issue? Are there additional adjustments or steps that I should consider? Your help on this matter would be greatly appreciated.

**Reply by ndr-brt** [reply]
are you using windows right? that's not specified but unfortunately they don't work on windows (the EDC is meant to run on a linux machine), this could be an issue to be added

**Reply by anstrasser** [reply]
Yes, I am using Windows. Thank you for the information. I'll try it on Linux at a later time.
